sdk/core/azure-core-http-httpurlconnection/src/main/java/com/azure/core/http/httpurlconnection/HttpUrlConnectionClient.java

```diff
@@ -0,0 +1,160 @@
+        return sendAsync(request, context);
+    }
+
+    // Synchronous send method with additional context, primarily for interface compliance
+    @Override
+    public HttpResponse sendSync(HttpRequest httpRequest, Context context) {
+        return sendAsync(httpRequest, context).block();
+    }
+
+    // Asynchronous send method with additional context
+    public Mono<HttpResponse> sendAsync(HttpRequest httpRequest, Context context) {
+        return openConnection(httpRequest)
+            .flatMap(connection -> {
+                HttpMethod httpMethod = httpRequest.getHttpMethod();
+
+                if (httpMethod == HttpMethod.PATCH) {
+                    return sendPatchViaSocket(httpRequest);
+                }
+
+                return setConnectionRequest(connection, httpRequest)
+                    .then(writeRequestBody(connection, httpRequest))
+                    .then(readResponse(connection, httpRequest))
+                    .doFinally(signalType -> connection.disconnect()) // Disconnect connection after processing
+                    .onErrorResume(e -> Mono.error(new RuntimeException(e)));
+            });
+    }
+
+    // Send a PATCH request via a SocketClient
+    private Mono<HttpResponse> sendPatchViaSocket(HttpRequest httpRequest) {
+        return Mono.fromRunnable(() -> {
+            try {
+                new SocketClient(httpRequest.getUrl().toString()).sendPatchRequest(httpRequest);
+            } catch (IOException e) {
+                throw new RuntimeException(e);
+            }
+        }).then(Mono.empty());
+    }
+
+    // Open a connection based on the HttpRequest URL
+    private Mono<HttpURLConnection> openConnection(HttpRequest httpRequest) {
+        return Mono.fromCallable(() -> {
+            URL url = new URL(httpRequest.getUrl().toString());
+            return (HttpURLConnection) url.openConnection();
+        });
+    }
+
+    // Set properties and headers on the HttpURLConnection
+    private Mono<Void> setConnectionRequest(HttpURLConnection connection, HttpRequest httpRequest) {
+        return Mono.fromRunnable(() -> {
+            try {
+                connection.setRequestMethod(httpRequest.getHttpMethod().toString());
+            } catch (ProtocolException e) {
+                throw new RuntimeException(e);
+            }
+            for (HttpHeader header : httpRequest.getHeaders()) {
+                connection.setRequestProperty(header.getName(), header.getValue());
+            }
+            if (httpRequest.getHttpMethod() != HttpMethod.GET)
+                connection.setDoOutput(true);
+        });
+    }
+
+    // Write the body of the request if necessary
+    private Mono<Void> writeRequestBody(HttpURLConnection connection, HttpRequest httpRequest) {
+        if (httpRequest.getHttpMethod() == HttpMethod.POST || httpRequest.getHttpMethod() == HttpMethod.PUT || httpRequest.getHttpMethod() == HttpMethod.PATCH) {
```

Comment:
From a style standpoing, I would rather we have an exhaustive switch statement here, e.g.

```
switch (httpRequest.getMethod()) {
    case POST:
    case PUT:
    case PATCH: {
        ...code
    }
    case <EVERYTHING ELSE>:
    default: return Mono.empty();
}